03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,33 +1,78 @@
 const Task = require('../models/Task')
-const getALLTasks = ( req,res) => {
-    res.send('Get all task')
-}
+const asyncWrapper = require('../middleware/async')
+const {createCustomError} = require('../errors/custom-error')
 
+const getALLTasks = asyncWrapper (async (req,res) => {
+ const tasks = await Task.find({})
+res.status(200).json({tasks})
+    
+    })
 
 
-const createTask = async (req, res) => {
-    const task = await Task.create(req.body)
-    res.status(201).json({task})
-}
 
-const getTask = (req, res) => {
-    res.json({id: req.params.id})
-}
 
-const updateTask = (req, res) => {
-    res.send('Update task')
-}
+const createTask = asyncWrapper (async (req, res) => {
+   
+        const task = await Task.create(req.body)
+        res.status(201).json({task}) 
+   
+    
+})
+
+const getTask = asyncWrapper (async (req, res, next) => {
+    
+        const {id:taskID} = req.params
+        const task = await Task.findOne({_id: taskID})
+
+        if(!task){
+            return next(createCustomError(`No task with id: ${taskID}`, 404 ))
+        
+        }
+        res.status(200).json({task})
+    
+    
+})
+
+
+
+const deleteTask = asyncWrapper(async (req, res) => {
```

Comment:
Requires change:
Missing `next` (`done`) parameter, therefore it'll be undefined below.

Reply:
oh, okay. I didn't put next because the instructor on the video didn't put next either and the instructions said we have to follow along.

Reply:
Where is it missing?

Reply:
On line 43, `next` doesn’t exist since it hasn’t been added a parameter to the `deleteTask` function!

```diff
- const deleteTask = asyncWrapper(async (req, res) => {
+ const deleteTask = asyncWrapper(async (req, res, next) => {
```